internal/controllers/module_reconciler.go

```diff
@@ -356,8 +356,10 @@ func (mrh *moduleReconcilerHelper) handleMIC(ctx context.Context, mod *kmmv1beta
 		images = append(images, mis)
 	}
 
-	if err := mrh.micAPI.CreateOrPatch(ctx, mod.Name, mod.Namespace, images, mod.Spec.ImageRepoSecret, mod); err != nil {
-		errs = append(errs, fmt.Errorf("failed to apply %s/%s MIC: %v", mod.Namespace, mod.Name, err))
+	if len(targetedNodes) > 0 {
```

Comment:
This solution is missing the following scenario:
1) Module is targeting some modules, MIC is created and with a number of images
2) user changes the NodeSelector field in Module
3) now Module is targeting zero nodes, but the images are not deleted from MIC since CreateOrPatch not being called

WDYT about the following solution:
initialize images in line 335 as:
image := []kmmv1beta1.ModuleImageSpec{} and always all CreateOrPatch function